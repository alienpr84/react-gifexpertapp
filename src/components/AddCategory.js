import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');

	const handleOnChangeInputValue = (e) => {
		setInputValue(e.target.value);
	}

	const handleOnSubmit = (e) => {
		e.preventDefault();
		
		if(inputValue.trim().length > 2) {
			setCategories( prev => [inputValue, ...prev]);
			setInputValue('');
		}
		
	}

	return (
		<div>
			<form onSubmit={handleOnSubmit}>
				<input 
					value= { inputValue }
					onChange={ handleOnChangeInputValue }
					placeholder="Enter a value"
				/>
			</form>
		</div>
	)
}

AddCategory.prototype = {
	setCategories: PropTypes.func.isRequired,
}