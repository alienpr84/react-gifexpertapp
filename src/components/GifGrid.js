import useFetchGifs from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs( category )

  return (
    <div>
      <h3>{category}</h3>

			{ loading && 'loading...' }

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem
						key={img.id}
						{...img}
					/>
        ))}
      </div>
    </div>
  );
};

export default GifGrid;
