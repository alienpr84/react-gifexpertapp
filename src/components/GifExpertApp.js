import { useState } from 'react';
import { AddCategory } from './AddCategory';
import GifGrid from './GifGrid';

const GifExpertApp = () => {
	const [categories, setCategories] = useState(['One punch']);

  return (
    <>
      <h2>GifExpertApp</h2>
			<AddCategory setCategories={ setCategories } />
      <hr />

			<ul>
			{
				categories.map( category => <GifGrid key={ category } category={ category }/>)
			}
			</ul>
    </>
  );
};

export default GifExpertApp;
