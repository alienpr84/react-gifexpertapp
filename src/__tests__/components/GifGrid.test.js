import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import useFetchGifs from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Testing on <GifGrid />', () => {
	const category = 'Naruto';

	test('Must show <GifGrid /> correctly', () => {
		useFetchGifs.mockReturnValue({ data: [], loading: true });
		const wrapper = shallow(<GifGrid category={ category } />)
		expect( wrapper ).toMatchSnapshot();
	});

	test('Must show data when the data is loaded', () => {
		const data = [
			{
				id: 'a1',
				url: 'http://www.google.com',
				username: 'username1',
				title: 'naruto 1',
			},
			{
				id: 'a2',
				url: 'http://www.google.com',
				username: 'username2',
				title: 'naruto 2',
			},
			{
				id: 'a3',
				url: 'http://www.google.com',
				username: 'username3',
				title: 'naruto 3',
			},
			{
				id: 'a4',
				url: 'http://www.google.com',
				username: 'username4',
				title: 'naruto 4',
			},
			{
				id: 'a5',
				url: 'http://www.google.com',
				username: 'username5',
				title: 'naruto 5',
			},
		];

		useFetchGifs.mockReturnValue({ 
			data, 
			loading: false,
		});

		const wrapper = shallow(<GifGrid category={ category } />);
		expect( wrapper ).toMatchSnapshot();
		expect( wrapper.find('p').exists() ).toBe(false);
		expect( wrapper.find('GifGridItem').length ).toBe( data.length );
	});
});