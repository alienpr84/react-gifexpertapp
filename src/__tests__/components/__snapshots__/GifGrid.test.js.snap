// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <GifGrid /> Must show <GifGrid /> correctly 1`] = `
<div>
  <h3>
    Naruto
  </h3>
  <p>
    loading...
  </p>
  <div
    className="card-grid"
  />
</div>
`;

exports[`Testing on <GifGrid /> Must show data when the data is loaded 1`] = `
<div>
  <h3>
    Naruto
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="a1"
      title="naruto 1"
      url="http://www.google.com"
      username="username1"
    />
    <GifGridItem
      id="a2"
      title="naruto 2"
      url="http://www.google.com"
      username="username2"
    />
    <GifGridItem
      id="a3"
      title="naruto 3"
      url="http://www.google.com"
      username="username3"
    />
    <GifGridItem
      id="a4"
      title="naruto 4"
      url="http://www.google.com"
      username="username4"
    />
    <GifGridItem
      id="a5"
      title="naruto 5"
      url="http://www.google.com"
      username="username5"
    />
  </div>
</div>
`;
